lntest/itest/lnd_routing_test.go

```diff
@@ -622,169 +620,106 @@ func testInvoiceRoutingHints(net *lntest.NetworkHarness, t *harnessTest) {
 	// First, we'll create a private channel between Alice and Bob. This
 	// will be the only channel that will be considered as a routing hint
 	// throughout this test. We'll include a push amount since we currently
-	// require channels to have enough remote balance to cover the invoice's
-	// payment.
-	chanPointBob := openChannelAndAssert(
-		t, net, net.Alice, net.Bob,
-		lntest.OpenChannelParams{
+	// require channels to have enough remote balance to cover the
+	// invoice's payment.
+	alice, bob := ht.Alice, ht.Bob
+	chanPointBob := ht.OpenChannel(
+		alice, bob, lntemp.OpenChannelParams{
 			Amt:     chanAmt,
 			PushAmt: chanAmt / 2,
 			Private: true,
 		},
 	)
 
-	// Then, we'll create Carol's node and open a public channel between her
-	// and Alice. This channel will not be considered as a routing hint due
-	// to it being public.
-	carol := net.NewNode(t.t, "Carol", nil)
-	defer shutdownAndAssert(net, t, carol)
+	// Then, we'll create Carol's node and open a public channel between
+	// her and Alice. This channel will not be considered as a routing hint
+	// due to it being public.
+	carol := ht.NewNode("Carol", nil)
 
-	net.ConnectNodes(t.t, net.Alice, carol)
-	chanPointCarol := openChannelAndAssert(
-		t, net, net.Alice, carol,
-		lntest.OpenChannelParams{
+	ht.ConnectNodes(alice, carol)
+	chanPointCarol := ht.OpenChannel(
+		alice, carol, lntemp.OpenChannelParams{
 			Amt:     chanAmt,
 			PushAmt: chanAmt / 2,
 		},
 	)
 
 	// We'll also create a public channel between Bob and Carol to ensure
-	// that Bob gets selected as the only routing hint. We do this as
-	// we should only include routing hints for nodes that are publicly
+	// that Bob gets selected as the only routing hint. We do this as we
+	// should only include routing hints for nodes that are publicly
 	// advertised, otherwise we'd end up leaking information about nodes
 	// that wish to stay unadvertised.
-	net.ConnectNodes(t.t, net.Bob, carol)
-	chanPointBobCarol := openChannelAndAssert(
-		t, net, net.Bob, carol,
-		lntest.OpenChannelParams{
+	ht.ConnectNodes(bob, carol)
+	chanPointBobCarol := ht.OpenChannel(
+		bob, carol, lntemp.OpenChannelParams{
 			Amt:     chanAmt,
 			PushAmt: chanAmt / 2,
 		},
 	)
 
-	// Then, we'll create Dave's node and open a private channel between him
-	// and Alice. We will not include a push amount in order to not consider
-	// this channel as a routing hint as it will not have enough remote
-	// balance for the invoice's amount.
-	dave := net.NewNode(t.t, "Dave", nil)
-	defer shutdownAndAssert(net, t, dave)
+	// Then, we'll create Dave's node and open a private channel between
+	// him and Alice. We will not include a push amount in order to not
+	// consider this channel as a routing hint as it will not have enough
+	// remote balance for the invoice's amount.
+	dave := ht.NewNode("Dave", nil)
```

Comment:
nit: do we want to defer shutting down Dave here?

Reply:
So a previous PR introduced the usage of `t.Cleanup`, which automatically cleans up the tests. The itest takes advantage of it, and auto shuts down non-standby nodes, that is, Alice and Bob, after each test is finished. This process is wrapped inside `HarnessTest.Subtest`. This way we can care less about the setup and cleanup and focus more on the test itself.